src/platform/macos/display.mm

```diff
@@ -37,38 +37,46 @@
     }
 
     capture_e
-    capture(snapshot_cb_t &&snapshot_cb, std::shared_ptr<img_t> img, bool *cursor) override {
-      __block auto img_next = std::move(img);
-
+    capture(const push_captured_image_cb_t &push_captured_image_cb, const pull_free_image_cb_t &pull_free_image_cb, bool *cursor) override {
       auto signal = [av_capture capture:^(CMSampleBufferRef sampleBuffer) {
-        auto av_img_next = std::static_pointer_cast<av_img_t>(img_next);
-
         CFRetain(sampleBuffer);
 
         CVPixelBufferRef pixelBuffer = CMSampleBufferGetImageBuffer(sampleBuffer);
         CVPixelBufferLockBaseAddress(pixelBuffer, kCVPixelBufferLock_ReadOnly);
 
-        if (av_img_next->pixel_buffer != nullptr)
-          CVPixelBufferUnlockBaseAddress(av_img_next->pixel_buffer, 0);
+        __block std::shared_ptr<img_t> img_out;
```

Comment:
`__block` isn't needed here. It's used if you're declaring something outside a block that you want to be able to modify from within the block (otherwise it would normally just be copied and changes wouldn't be reflected in the original).

Reply:
Fixed.